### PR TITLE
Fix broken links

### PR DESCRIPTION
**Pull request by danelson**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet/issues/5030

## Changes

Updated broken links that had a base uri of https://github.com/open-telemetry/opentelemetry-specification

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 1]
```diff
@@ -11,9 +11,9 @@ also collects traces from incoming gRPC requests using
 [Grpc.AspNetCore](https://www.nuget.org/packages/Grpc.AspNetCore).
 
 **Note: This component is based on the OpenTelemetry semantic conventions for
-[metrics](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions)
+[metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md)
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:14]:
```suggestion
[metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md)
```

File: src/OpenTelemetry.Instrumentation.AspNetCore/README.md [hunk 2]
```diff
@@ -11,9 +11,9 @@ also collects traces from incoming gRPC requests using
...
 and
-[traces](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions).
+[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/trace.md).
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.AspNetCore/README.md:16]:
```suggestion
[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md).
```

File: src/OpenTelemetry.Instrumentation.GrpcNetClient/README.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ which instruments [Grpc.Net.Client](https://www.nuget.org/packages/Grpc.Net.Clie
 and collects traces about outgoing gRPC requests.
 
 **Note: This component is based on the OpenTelemetry semantic conventions for
-[traces](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions).
+[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/trace.md).
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.GrpcNetClient/README.md:11]:
```suggestion
[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/rpc/rpc-spans.md).
```

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 4]
```diff
@@ -12,9 +12,9 @@ and
 and collects metrics and traces about outgoing HTTP requests.
 
 **Note: This component is based on the OpenTelemetry semantic conventions for
-[metrics](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions)
+[metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md)
 and
-[traces](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions).
+[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/trace.md).
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.Http/README.md:17]:
```suggestion
[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md).
```

File: src/OpenTelemetry.Instrumentation.Http/README.md [hunk 5]
```diff
@@ -12,9 +12,9 @@ and
 and collects metrics and traces about outgoing HTTP requests.
 
 **Note: This component is based on the OpenTelemetry semantic conventions for
-[metrics](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions)
+[metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md)
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.Http/README.md:15]:
```suggestion
[metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-metrics.md)
```

File: src/OpenTelemetry.Instrumentation.SqlClient/README.md [hunk 6]
```diff
@@ -18,7 +18,7 @@ and later.
 >
 > **Note**
 > This component is based on the OpenTelemetry semantic conventions for
-[traces](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions).
+[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/trace.md).
```
Comment (vishweshbankwar) [inline src/OpenTelemetry.Instrumentation.SqlClient/README.md:21]:
```suggestion
[traces](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md).
```

**Review by vishweshbankwar — APPROVED** [review]
Thanks @danelson!

**Comment by danelson** [comment]
@vishweshbankwar no problem. Thanks for your review and good idea to use the domain specific links where applicable. I added your changes.

**Review by utpilla — APPROVED** [review]

**Pull request merged**